### DISCUSSION TITLE
Z axis limit switches

### DISCUSSION THREAD

**Discussion by cpapplefamily** [grblHAL/core#370] [Q&A] [answered]
I have a picoCNC running the grblhal firmware.  The board has limit switche accommodations for A,X,Y,Z. I'm currently only using X,Y,Z for a laser table.  I would like Z to have both positive and negative limit switches.

Can someone point me to an example to how of possible to set this up? 

We will likely be adding a rotary axis on the future so I will likely enable Axis A bit will not need the limit

**Comment by terjeio** [comment, marked as answer]
The RP2040 driver currently does not have support for max limit switches so creating a custom map file is not an option.
Easiest is to connect both switches to the Z limit input, normally closed (NC) switches in series, normall open (NO) in paralell. There is a small problem with this though, if homing is initiated with one of the switches triggered the homing routine cannot know which direction to pulloff. To avoid a crash in this scenario "Two switches shares one input pin" (bit 4) in the `$22` setting should be enabled.
`$$=22` output:
```
    0 - Enable (1)
    1 - Enable single axis commands (2)
    2 - Homing on startup required (4)
    3 - Set machine origin to 0 (8)
    4 - Two switches shares one input pin (16)
    5 - Allow manual (32)
    6 - Override locks (64)
    7 - Keep homed status on reset (128)
```

**Reply by cpapplefamily** [reply]
Thanks for the great answer. BTW where did you get that Bit Map? The [https://github.com/terjeio/grblHAL/blob/master/doc/markdown/settings.md](https://github.com/terjeio/grblHAL/blob/master/doc/markdown/settings.md) does an ok job with many of the settings but this particular one it does not

**Reply by terjeio** [reply]
It is not a bitmap, it is text from the controller copied from the output returned from sending `$$=22` to the controller.
See the [Wiki for new commands](https://github.com/grblHAL/core/wiki/Additional-or-extended-settings) that can be used to get settings-related information from the controller. BTW your link above is to the old grblHAL repository which is now way out of date.
